src/marker.rs

```diff
@@ -951,7 +948,14 @@ impl<'py> Python<'py> {
     /// # Safety
     ///
     /// Callers must ensure that ensure that the cast is valid.
-    #[allow(clippy::wrong_self_convention)]
+    #[allow(clippy::wrong_self_convention, deprecated)]
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "use `Py::from_borrowed_ptr(py, ptr)` or `Bound::from_borrowed_ptr(py, ptr)` instead"
```

Comment:
I think _technically_ the equivalent method would be on `Borrowed`, but I thought we might want to nudge user here to the safer option.

Reply:
Agreed I think `Borrowed::from_ptr` is hard to get right, as we keep finding by not using it in our own macro code!